lib/engines/hcktest/platforms/schema.json

```diff
@@ -0,0 +1,130 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "name": {
+      "type": "string"
+    },
+    "kit": {
+      "type": "string"
+    },
+    "type": {
+      "type": "string",
+      "enum": [
+        "virtual",
+        "physical"
+      ]
+    },
+    "cpu": {
+      "type": "string"
+    },
+    "machine_type": {
+      "enum": [
+        "q35",
+        "pc"
+      ]
+    },
+    "fw_type": {
+      "enum": [
+        "uefi",
+        "bios"
+      ]
+    },
+    "viommu_state": {
+      "type": "boolean"
+    },
+    "enlightenments_state": {
+      "type": "boolean"
+    },
+    "device_extra": {
+      "type": "string"
+    },
+    "st_image": {
+      "type": "string"
+    },
```

Comment:
Shouldn't these be moved into the `if`? Otherwise a physical platform can specify e.g., `q35` for `machine_type`, but it's clearly wrong.